README_docker.md

```diff
@@ -0,0 +1,164 @@
+
+This generates keys in `~/.drand/keys/` on the host.
+
+## Docker & Docker-compose setup
+
+Create a `Dockerfile` with the following contents:
+
+	FROM golang:1.12.5-alpine # update this number when possible, but keep "alpine" for a lightweight image
+	RUN apk update
+	RUN apk upgrade
+	RUN apk add git
+	RUN go get -u github.com/dedis/drand
+	WORKDIR /
+	ENTRYPOINT ["drand", "--verbose", "2", "start", "--listen", "0.0.0.0:8080"]
+
+**Note:** if you intend to use `drand` behind a reverse-proxy, add `--tls-disable` here; please read "Docker behind reverse proxy setup" at the end of this guide. 
+
+Let's continue with docker-compose. It will help keeping our container running, plus manage volumes, etc.
+
+Create the following `docker-compose.yml` :
+
+	drand:
+	  build: .
+	  volumes:
+	    - ./data:/root/.drand
+	  ports:
+	    - "0.0.0.0:1234:8080"
+
+Finally, also create a `data` folder to hold the keys and settings. It __HAS__ to have permissions `740`.
+
+	mkdir data
+	chmod 740 data
+
+Drand can then be built into an image with
+
+	docker-compose up --build
+
+Finally, the image is ready for DKG. Just start it and let it run in background with 
+
+	docker-compose up -d
+
+(one can also set a restart policy in the docker-compose.yml)
+
+## DKG
+
+If you did the setup above, you have a container running, loaded with your keys. It still misses two things:
+
+1. the group.toml file corresponding to other participants. For this, you have to exchange keys manually, e.g., via email.
+2. running the DKG protocol to bootstrap drand.
+
+Fortunately, with our docker-compose volumes, it's now very easy to add things into the running container. Just add your `group.toml` into the root of the `data` folder (__NOT__ in the `data/groups/` folder; this one is manually managed by drand, don't touch it).
+
+Then, open a CLI into your running docker.
+
+First find its id on the host:
+
+	$ docker ps
+	697e4766f8b2        drand_drand             "drand --verbose 2 s…"   11 minutes ago      Up 9 minutes
+
+The id of the container is `697e4766f8b2`. Enter it by running:
+
+	docker exec -it 697e4766f8b2 /bin/sh
+
+Then, you're inside the container; tell drand to run the DKG like so:
+
+	drand share /root/.drand/group.toml
+
+__Notice the full path__ `/root/.drand/group.toml` and not `group.toml` nor `./group.toml`
+
+## Checking the logs
+
+From outside the container, run `docker-compose logs`
+
+## Updating drand
+
+To update the docker, simply shut it down
+
+	docker-compose down
+
+and to fully rebuild it, you need to first clean the already-used layers (for some reason Docker is confused and thinks nothing has changed, and it keeps rebuilding the old version)
+
+	docker system prune -a
+
+Caution! this delete *all* your *unused* containers and networks; it's typically fine, but just be aware of it.
+
+Then rebuild and restart it
+
+	docker-compose up --build -d
+
+## Something went wrong, reset (without losing the keys)
```

Comment:
=> "Reset drand state" seems more concise ?
And having a first sentence that says why and what it does, stg like "if you need to reset drand's internal state but keeping drand private key, you can do so by manually deleting specific folders in the configuration folder of drand" . blablabla